pkg/gather/clusterconfig/version.go

```diff
@@ -2,75 +2,110 @@ package clusterconfig
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/rest"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
 
 	configv1 "github.com/openshift/api/config/v1"
 	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
 
 	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils/check"
 	"github.com/openshift/insights-operator/pkg/utils/marshal"
 	"github.com/openshift/insights-operator/pkg/utils/anonymize"
 )
 
-// GatherClusterVersion fetches the ClusterVersion - the ClusterVersion with name version.
+// GatherClusterVersion fetches the ClusterVersion with the name 'version' and its resources.
```

Comment:
It would be nice to mention the cluster ID here at the beginning IMO.

Reply:
Done with  8699181 -> 518c25a.